iphone/Maps/UI/Editor/MWMEditorViewController.mm

```diff
@@ -724,6 +724,18 @@ - (void)fillCell:(UITableViewCell * _Nonnull)cell atIndexPath:(NSIndexPath * _No
   }
 }
 
+- (void)removeAdditionalNameAtIndexPath:(NSIndexPath *)indexPath
+{
+  MWMEditorAdditionalNameTableViewCell * cell = [self.tableView cellForRowAtIndexPath:indexPath];
+  if (!cell)
+    return;
+  [self cell:cell changedText:@""];
```

Comment:
Is this call needed? Does it indirectly clear the translation underneath?

Reply:
All empty names will be cleaned up automatically by the core:
```objc
- (void)removeAdditionalNameAtIndexPath:(NSIndexPath *)indexPath
{
  MWMEditorAdditionalNameTableViewCell * cell = [self.tableView cellForRowAtIndexPath:indexPath];
  if (!cell)
    return;
  [self cell:cell changedText:@""];
  m_mapObject.RemoveNeedlessNames(); // <---
```

The previous implementation doesn't support the direct languages removal and asserts when the string is nil:
```obj
- (void)cell:(MWMTableViewCell *)cell changedText:(NSString *)changeText
{
  NSAssert(changeText != nil, @"String can't be nil!");
```

So when non-empty language is deleted, first of all, we should empty the name.